examples/structured_data/classification_with_grn_and_vsn.py

```diff
@@ -0,0 +1,413 @@
+"""
+Title: Classification with Gated Residual and Variable Selection Networks
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/01/15
+Last modified: 2021/01/15
+Description: Using Gated Residual and Variable Selection Networks for income level prediction.
+"""
+
+"""
+## Introduction
+
+This example demonstrates the use of Gated
+Residual Network (GRN) and Variable Selection Networks (VSN), proposed by
+[Bryan Lim et al.](https://arxiv.org/abs/1912.09363), for structured data
+classification. GRNs give the flexibility to the model to apply non-linear
+processing only where needed. VSNs allow the model to softly remove any
+unnecessary noisy inputs which could negatively impact performance.
+Together, those techniques help improving the learning capacity of deep neural
+network models.
+
+
+Note that this example should be run with TensorFlow 2.3 or higher.
```

Comment:
Suggestion:

- Try to avoid repetition ("Note that this example... [Headline]... This examples uses... ").

```suggestion
Note that to run the code you need to use TensorFlow 2.3 or higher.
```